org.jcryptool.visual.rabin/src/org/jcryptool/visual/rabin/DataTransfer.java

```diff
@@ -0,0 +1,327 @@
+		rstc.txtCipherFirst.setText(state.get(0));
+		rstc.txtmp.setText(state.get(1));
+		rstc.txtmq.setText(state.get(2));
+		rstc.txtyp.setText(state.get(3));
+		rstc.txtyq.setText(state.get(4));
+		rstc.txtV.setText(state.get(5));
+		rstc.txtW.setText(state.get(6));
+		rstc.txtm1.setText(state.get(7));
+		rstc.txtm2.setText(state.get(8));
+		rstc.txtm3.setText(state.get(9));
+		rstc.txtm4.setText(state.get(10));
+		
+	}
+	
+	
+	
+	
+	/**
+	 * save chooseCipherIdx in text mode
+	 * @param rstc
+	 */
+	private void saveChooseCipherIdxTextMode(RabinSecondTabComposite rstc) {
+		cmbChooseCipherIdxTextMode = rstc.cmbChooseCipher.getSelectionIndex();
+	}
+	
+	/**
+	 * save chooseCipherIdx in num mode
+	 * @param rstc
+	 */
+	private void saveChooseCipherIdxNumMode(RabinSecondTabComposite rstc) {
+		cmbChooseCipherIdxNumMode = rstc.cmbChooseCipher.getSelectionIndex();
+	}
+	
+	/**
+	 * save chooseCipherIdx in decryption hex mode
+	 * @param rstc
+	 */
+	private void saveChooseCipherIdxDecHexMode(RabinSecondTabComposite rstc) {
+		cmbChooseCipherIdxDecHexMode = rstc.cmbChooseCipher.getSelectionIndex();
+	}
+	
+	/**
+	 * save chooseCipher in decryption decimal mode
+	 * @param rstc
+	 */
+	private void saveChooseCipherIdxDecDecimalMode(RabinSecondTabComposite rstc) {
+		cmbChooseCipherIdxDecDecimalMode = rstc.cmbChooseCipher.getSelectionIndex();
+	}
+	
+	/**
+	 * set chooseCipher num mode
+	 * @param rstc
+	 */
+	private void setChooseCipherIdxNumMode(RabinSecondTabComposite rstc) {
+		if(cmbChooseCipherIdxNumMode != -1) 
+			rstc.cmbChooseCipher.select(cmbChooseCipherIdxNumMode);
+	}
+	
+	/**
+	 * set chooseCipher text mode
+	 * @param rstc
+	 */
+	private void setChooseCipherIdxTextMode(RabinSecondTabComposite rstc) {
+		if(cmbChooseCipherIdxTextMode != -1) 
+			rstc.cmbChooseCipher.select(cmbChooseCipherIdxTextMode);
+	}
+	
+	/**
+	 * set chooseCipher decryption hex mode
+	 * @param rstc
+	 */
+	private void setChooseCipherIdxDecHexMode(RabinSecondTabComposite rstc) {
+		if(cmbChooseCipherIdxDecHexMode != -1) 
+			rstc.cmbChooseCipher.select(cmbChooseCipherIdxDecHexMode);
+	}
+	
+	/**
+	 * set chooseCipher decryption decimal mode
+	 * @param rstc
+	 */
+	private void setChooseCipherIdxDecDecimalMode(RabinSecondTabComposite rstc) {
+		if(cmbChooseCipherIdxDecDecimalMode != -1) 
+			rstc.cmbChooseCipher.select(cmbChooseCipherIdxDecDecimalMode);
+	}
+	
+	
+	
+	
+	
+	
+	
+	
+	
+	
+	
+	/**
+	 * set chooseCipher by filling combo and setting idx
+	 * @param a
+	 * @param rstc
+	 */
+	public void setChooseCipher(ArrayList<String> a, RabinSecondTabComposite rstc) {
+		rstc.cmbChooseCipher.removeAll();
+		if(a != null) {
+			for(int i = 1; i <= a.size(); i++) {
+				rstc.cmbChooseCipher.add(String.valueOf(i));
+			}
+		}
+	}
+	
+	
+	
+
+	
+	
+	/**
+	 * save num state
+	 * @param rstc
+	 */
+	public void saveNumState(RabinSecondTabComposite rstc) {
+		saveChooseCipherIdxNumMode(rstc);
+		String[] decState = getDecState(rstc);
+		saveState(stateNum, decState);
```

Comment:
As long as it is not too complex, it pays off to do things as this. Shorter files reduce cognitive load.
```suggestion
		saveState(stateNum, getDecState(rstc));
```